src/AppInstallerCommonCore/UserSettings.cpp

```diff
@@ -0,0 +1,235 @@
+        }
+
+        std::optional<Json::Value> ParseFile(const std::filesystem::path& fileName, std::vector<std::string>& warnings)
+        {
+            auto stream = GetSettingStream(Type::UserFile, fileName);
+            if (stream)
+            {
+                Json::Value root;
+                Json::CharReaderBuilder builder;
+                const std::unique_ptr<Json::CharReader> reader(builder.newCharReader());
+
+                std::string settingsContentStr = Utility::ReadEntireStream(*stream);
+                std::string error;
+
+                if (reader->parse(settingsContentStr.c_str(), settingsContentStr.c_str() + settingsContentStr.size(), &root, &error))
+                {
+                    return root;
+                }
+
+                warnings.emplace_back(fileName.u8string());
+                warnings.emplace_back(error);
+            }
+
+            return {};
+        }
+
+        template <Setting S>
+        void Validate(
+            Json::Value& root,
+            std::map<Setting, details::SettingVariant>& settings,
+            std::vector<std::string>& warnings)
+        {
+            // jsoncpp doesn't support std::string_view yet.
+            auto path = std::string(details::SettingMapping<S>::Path);
+
+            const Json::Path jsonPath(path);
+            Json::Value result = jsonPath.resolve(root);
+            if (!result.isNull())
+            {
+                auto jsonValue = GetValue<details::SettingMapping<S>::json_t>(result);
+
+                if (jsonValue.has_value())
+                {
+                    auto validatedValue = details::SettingMapping<S>::Validate(jsonValue.value());
+
+                    if (validatedValue.has_value())
+                    {
+                        // Finally add it to the map
+                        settings[S].emplace<details::SettingIndex(S)>(
+                            std::forward<typename details::SettingMapping<S>::value_t>(validatedValue.value()));
+                    }
+                    else
+                    {
+                        warnings.push_back(GetSettingsMessage(SettingsWarnings::InvalidFieldValue, path, jsonValue.value()));
+                    }
+                }
+                else
+                {
+                    warnings.emplace_back(GetSettingsMessage(SettingsWarnings::InvalidFieldFormat, path));
+                }
+            }
+        }
+
+        template <size_t... S>
+        void ValidateAll(
+            Json::Value& root,
+            std::map<Setting, details::SettingVariant>& settings,
+            std::vector<std::string>& warnings,
+            std::index_sequence<S...>)
+        {
+            // Use folding to call each setting validate function.
+            // Do not change this expression without understanding the implications to the bind order.
+            // See: https://en.cppreference.com/w/cpp/language/fold for more details.
```

Comment:
These lines are not relevant; order should not matter here and it certainly doesn't have anything to do with binding (that is SQLite related, no C++).